components/script/dom/url.rs

```diff
@@ -105,6 +112,90 @@ impl URL {
     pub fn DomainToUnicode(_: GlobalRef, origin: USVString) -> USVString {
         USVString(domain_to_unicode(&origin.0))
     }
+
+    // https://w3c.github.io/FileAPI/#dfn-createObjectURL
+    pub fn CreateObjectURL(global: GlobalRef, blob: &Blob) -> DOMString {
+        /// XXX: Second field is an unicode-serialized Origin, it is a temporary workaround
+        ///      and should not be trusted. See issue https://github.com/servo/servo/issues/11722
+        let origin = global.get_url().origin().unicode_serialization();
+
+        if blob.IsClosed() {
+            // Generate a dummy id
+            let id = Uuid::new_v4().simple().to_string();
+            return DOMString::from(URL::unicode_serialization_blob_url(&origin, &id));
+        }
+
+        let filemanager = global.resource_threads().sender();
+
+        let slice = blob.get_slice_or_empty();
+        let bytes = slice.get_bytes();
+
+        let entry = BlobURLStoreEntry {
+            type_string: blob.Type().to_string(),
+            filename: None, // XXX: the filename is currently only in File object now
+            size: blob.Size(),
+            bytes: bytes.to_vec(),
+        };
+
+        let (tx, rx) = ipc::channel().unwrap();
+
+        let msg = BlobURLStoreMsg::AddEntry(entry, origin.clone(), tx);
+
+        let _ = filemanager.send(FileManagerThreadMsg::BlobURLStoreMsg(msg));
+
+        match rx.recv().unwrap() {
+            Ok(id) => {
+                DOMString::from(URL::unicode_serialization_blob_url(&origin, &id))
+            }
+            Err(_) => {
+                // Generate a dummy id
+                let id = Uuid::new_v4().simple().to_string();
+                DOMString::from(URL::unicode_serialization_blob_url(&origin, &id))
+            }
+        }
+    }
+
+    // https://w3c.github.io/FileAPI/#dfn-revokeObjectURL
+    pub fn RevokeObjectURL(global: GlobalRef, url: DOMString) {
+        /*
+            If the url refers to a Blob that has a readability state of CLOSED OR
+            if the value provided for the url argument is not a Blob URL, OR
+            if the value provided for the url argument does not have an entry in the Blob URL Store,
+
+            this method call does nothing. User agents may display a message on the error console.
+
+            XXX: First condition can't be enforced given current design
```

Comment:
Why not? 

Anyway, this step seems to already be  run on the fm thread, sort of.